cleanlab/dataset.py

```diff
@@ -0,0 +1,416 @@
+        * Overall Label Noise Score
+            prob(true_label != k | given_label = k)
+            Estimated proportion of label issues in the class.
+        * Inverse Label Noise Score
+            prob(given_label != k | true_label = k)
+            Estimated proportion of times this class label occurs in other classes in the dataset.
+        * Overall Label Quality Score
+            p(given_label = k | true_label = k)
+            How accurate the labels are in each class as a conditional prob p(given=k | true=k).
+        By default, the DataFrame is ordered by "Overall Label Noise Score" descending.
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+    given_label_noise = joint.sum(axis=1) - joint.diagonal()  # p(s=k) - p(s=k,y=k) = p(y!=k, s=k)
+    true_label_noise = joint.sum(axis=0) - joint.diagonal()  # p(y=k) - p(s=k,y=k) = p(s!=k,y=k)
+    given_conditional_noise = given_label_noise / joint.sum(axis=1)  # p(y!=k, s=k) / p(s=k)
+    true_conditional_noise = true_label_noise / joint.sum(axis=0)  # p(s!=k, y=k) / p(y=k)
+    df = pd.DataFrame(
+        {
+            "Class Index": np.arange(len(joint)),
+            "Overall Label Issues": (given_label_noise * num_examples).round().astype(int),
+            "Inverse Label Issues": (true_label_noise * num_examples).round().astype(int),
+            "Overall Label Noise Score": given_conditional_noise,  # p(y!=k | s=k)
+            "Inverse Label Noise Score": true_conditional_noise,  # p(s!=k | y=k)
+            # Below could equivalently be computed as: joint.diagonal() / joint.sum(axis=1)
+            "Overall Label Quality Score": 1 - given_conditional_noise,  # p(y=k | s=k)
+        }
+    )
+    if class_names is not None:
+        df.insert(loc=0, column="Class Name", value=class_names)
+    return df.sort_values(by="Overall Label Noise Score", ascending=False)
+
+
+def find_overlapping_classes(
+    labels=None,
+    pred_probs=None,
+    *,
+    asymmetric=False,
+    class_names=None,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Returns the classes that are often confused by machine learning model or data labelers.
+    Consider merging the top pairs of classes returned by this method each into a single class.
+    If the dataset is labeled by human annotators, consider clearly defining the
+    difference between the classes prior to have annotators label the data.
+
+    This method provides two scores in the pandas Data Frame that is returned:
+    * "Num Overlapping Examples" - The number of examples where the two classes overlap
+    * "Joint Probability" - "Num Overlapping Examples" / total number of examples in the dataset.
```

Comment:
I like the score "joint probability" because it's very interpretable. However one issue is this violates the convention of our package that lower scores => more serious problem. 

If we want to be consistent with this convention (which I advise given we plan to have tons of data-quality scores in the package), then alternate option is to call this "Overlap Score" and in the docstring define it as 1 - joint probability, i.e. = 1 - (Number of overlapping examples) / (Total number of examples in the dataset)